File: ppbluetoothkit/src/main/java/com/peng/ppscale/business/v4/Md5Util.java
package com.peng.ppscale.business.v4;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class Md5Util {


    private static final char[] hexCode = "0123456789ABCDEF".toCharArray();

    public static byte[] md5Bytes(String input) {
        byte[] bytes = new byte[0];
        try {
            bytes = MessageDigest.getInstance("MD5").digest(input.getBytes());
            return bytes;
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String md5(String input) {
        byte[] bytes = new byte[0];
        try {
            bytes = MessageDigest.getInstance("MD5").digest(input.getBytes());
            return printHexBinary(bytes);
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String printHexBinary(byte[] data) {
        StringBuilder r = new StringBuilder(data.length * 2);
        for (byte b : data) {
            r.append(hexCode[(b >> 4) & 0xF]);
            r.append(hexCode[(b & 0xF)]);
        }
        return r.toString();
    }

}

File: ppbluetoothkit/src/main/java/com/peng/ppscale/key/MD5Util.kt
package com.peng.ppscale.key

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object MD5Util {
    /**
     * Md5加密
     *
     * @param message
     * @return
     */
    fun md5Encoder(message: String): String {
        return try {
            val digest = MessageDigest.getInstance("md5")
            val encryption = digest.digest(message.toByteArray())
            val hexValue = StringBuffer()
            for (i in encryption.indices) {
                val `val` = encryption[i].toInt() and 0xff
                if (`val` < 16) {
                    hexValue.append("0")
                }
                hexValue.append(Integer.toHexString(`val`))
            }
            hexValue.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
            ""
        }
    }
}